File: solution/day04/Routine.Tests/Fakes/ScheduleService.cs
using FluentAssertions;

namespace Routine.Tests.Fakes;

public class ScheduleService(Schedule schedule) : IScheduleService
{
    private Schedule? _calledToOrganize;
    private bool _continue;

    public Schedule TodaySchedule() => schedule;

    public void OrganizeMyDay(Schedule s) => _calledToOrganize = s;

    public void Continue() => _continue = true;

    public void HasBeenOrganized(Schedule s) => _calledToOrganize.Should().Be(s);
    public void HasBeenContinued() => _continue.Should().BeTrue();
}

File: exercise/ts/day04/tests/doubles/scheduleService.ts
import { ScheduleService } from '../../src/routine/scheduleService';
import { Schedule } from '../../src/routine/schedule';

export class ScheduleServiceDouble implements ScheduleService{
    todaySchedule() {
        return new Schedule();
    }

    private _organizeMyDayCalled = false;
    get organizeMyDayCalled() {
        return this._organizeMyDayCalled;
    }
    organizeMyDay(schedule: Schedule) {
        this._organizeMyDayCalled = true;
    }

    private _continueDayCalled = false;
    get continueDayCalled() {
        return this._continueDayCalled;
    }
    continueDay() {
        this._continueDayCalled = true;
    }
}
